saltproc/saltproc.py

```diff
@@ -55,103 +71,280 @@ def __init__(self, steps, cores, nodes, bw, exec_path, restart=False,
-        self.bu_adens_db_0 = self.f.create_dataset('core adensity before reproc',
-                                                   shape, maxshape=maxshape,
-                                                   chunks=True)
-        self.bu_adens_db_1 = self.f.create_dataset('core adensity after reproc',
-                                                   shape, maxshape=maxshape,
-                                                   chunks=True)
-        self.tank_adens_db = self.f.create_dataset('tank adensity',
+        self.driver_before_db = self.f.create_dataset('driver composition before reproc',
+                                                      shape, maxshape=maxshape,
+                                                      chunks=True)
+        self.driver_after_db = self.f.create_dataset('driver composition after reproc',
+                                                     shape, maxshape=maxshape,
+                                                     chunks=True)
+        self.driver_refill_tank_db = self.f.create_dataset('driver refill tank composition',
+                                                           shape, maxshape=maxshape,
+                                                           chunks=True)
+
+        self.blanket_before_db = self.f.create_dataset('blanket composition before reproc',
+                                                       shape, maxshape=maxshape,
+                                                       chunks=True)
+        self.blanket_after_db = self.f.create_dataset('blanket composition after reproc',
+                                                      shape, maxshape=maxshape,
+                                                      chunks=True)
+        self.blanket_refill_tank_db = self.f.create_dataset('blanket refill tank composition',
+                                                            shape, maxshape=maxshape,
+                                                            chunks=True)
+
+        self.fissile_tank_db = self.f.create_dataset('fissile tank composition', shape, maxshape=maxshape,
+                                                     chunks=True)
+        self.waste_tank_db = self.f.create_dataset('waste tank composition',
                                                    shape, maxshape=maxshape,
                                                    chunks=True)
-        self.noble_adens_db = self.f.create_dataset('noble adensity',
-                                                    shape, maxshape=maxshape,
-                                                    chunks=True)
-        self.th_adens_db = self.f.create_dataset('Th tank adensity',
-                                                 shape, maxshape=maxshape,
-                                                 chunks=True)
-        # !! raffinate stream consider splitting by what element
-        self.rem_adens = np.zeros((5, self.number_of_isotopes))
+        # initial composition
```

Comment:
no comment needed. If a comment is needed, probably this should be its own function.